src/main/java/org/spongepowered/common/mixin/core/entity/MixinEntity.java

```diff
@@ -774,6 +776,34 @@ public DataTransactionResult addPassenger(Entity entity) {
     }
 
     @Override
+    public DataTransactionResult removePassenger(Entity entity) {
+        checkNotNull(entity);
+        if (entity.getPassengers().contains(this)) {
+            throw new IllegalArgumentException(String.format("Cannot remove entity %s as a passenger of %s, because the former is not a passenger "
+                    + "of the latter ", entity, this));
+        }
+
+        final DataTransactionResult.Builder builder = DataTransactionResult.builder();
+        this.removePassenger((net.minecraft.entity.Entity) entity);
+
+        final ImmutableList.Builder<EntitySnapshot> passengerSnapshotsBuilder = ImmutableList.builder();
+
+        return builder.result(DataTransactionResult.Type.SUCCESS).success(new ImmutableSpongeListValue<>(Keys.PASSENGERS, passengerSnapshotsBuilder
+                .build())).build();
+    }
+
+    @Override
+    public DataTransactionResult clearPassengers() {
+        this.removePassengers();
+
+        final ImmutableList.Builder<EntitySnapshot> passengerSnapshotsBuilder = ImmutableList.builder(); // This is an empty list. No passengers.
```

Comment:
From what I can tell, empty - because it's returning the data after it's been changed. I'm not sure if this is right though.